src/core/qgsvectorlayerutils.h

```diff
@@ -68,6 +85,14 @@ class CORE_EXPORT QgsVectorLayerUtils
                                      const QgsAttributeMap &attributes = QgsAttributeMap(),
                                      QgsExpressionContext *context = nullptr );
 
+    /**
+     * Duplicates a feature and it's children (one level deep). It calls CreateFeature, so
+     * default values and constraints (e.g., unique constraints) will automatically be handled.
+     * The duplicated feature will be automatically inserted into the layer. Or not?
```

Comment:
Not sure the API documentation is the right place to ask questions, I normally expect it to answer questions ;)

Reply:
Btw, all neew methods will need to have an additional

```
\since QGIS 3.0
```

Reply:
Also if the  class is new?
And variables too?